ipatests/test_integration/test_nfs.py

```diff
@@ -19,249 +16,473 @@
 import re
 import time
 
+import pytest
+
 from ipatests.test_integration.base import IntegrationTest
 from ipatests.pytest_ipa.integration import tasks
+from ipatests.pytest_ipa.integration.firewall import Firewall
+from ipaplatform import services
+from ipaplatform.paths import paths
 
 # give some time for units to stabilize
 # otherwise we get transient errors
 WAIT_AFTER_INSTALL = 5
 WAIT_AFTER_UNINSTALL = WAIT_AFTER_INSTALL
 
-
-class TestNFS(IntegrationTest):
-
-    num_clients = 3
-    topology = 'line'
-
-    def cleanup(self):
-
-        nfssrv = self.clients[0]
-        nfsclt = self.clients[1]
-        automntclt = self.clients[2]
-
+LOCATION = "seattle"
+IPA_USER = "athena"
+IPA_PASSWORD = "Secret123"
+
+NFS_EXPORTS = {
+    'krbnfs_manual': {
+        'export_options': '*(sec=krb5p,no_subtree_check,rw)',
+        'export_rpath': 'krbnfs_manual',
+        'exposed_path': '/krbnfs_manual',
+        'mount_path': '/mnt/krbnfs_manual',
+    },
+    'krbnfs_autofs': {
+        'export_options': '*(sec=krb5p,no_subtree_check,rw)',
+        'export_rpath': 'krbnfs_autofs',
+        'exposed_path': '/krbnfs_autofs',
+        'mount_path': '/mnt/krbnfs_autofs',
+    },
+    'krbnfs_home': {
+        'export_options': '*(sec=krb5p,no_subtree_check,rw)',
+        'export_rpath': os.path.join('krbnfs_home', IPA_USER),
+        'exposed_path': os.path.join(os.sep, 'krbnfs_home', IPA_USER),
+        'mount_path': os.path.join('/home', IPA_USER),
+    },
+}
+
+
+@pytest.fixture(scope="class")
+def krb5_nfs_client(request, mh):
```

Comment:
TestNFS::test_automount used to test ipa-client-automount on unmodified systems.
This is important as testing ipa-client-automount on unmodified systems caught quite a few bugs.
This is not done anymore as krb5_nfs_client is called before ipa-client-automount.